src/main/schema.ts

```diff
@@ -1,29 +1,19 @@
 export const toggleSchema: object = {
     type: 'object',
-    properties: {
-        toggles: {
-            type: 'array',
-            items: {
-                type: 'object',
-                properties: {
-                    id: {
-                        type: 'string',
-                    },
-                    description: {
-                        type: 'string',
-                    },
-                    fraction: {
-                        type: 'number',
-                        minimum: 0.0,
-                        maximum: 1.0,
-                    },
-                    comment: {
-                        type: 'string',
-                    },
+    patternProperties: {
+        '^[A-Za-z0-9 -_.]': {
+            type: 'object',
+            properties: {
+                fraction: {
+                    type: 'number',
+                    minimum: 0.0,
+                    maximum: 1.0,
+                },
+                type: {
+                    type: 'string',
```

Comment:
Would there be any thought to making this an enum like ["BOOL", "RAMP"]? Or would the client need to have to be able to define this?

Reply:
Not sure how this can be achieved. But will be good to have that check.